src/megaclient.cpp

```diff
@@ -926,12 +934,16 @@ void MegaClient::exec()
 
                         if (!fc->timeout.armed()) break;
 
+                        LOG_warn << "Timeout getting file attr";
                         // timeout! fall through...
-                        fc->req.disconnect();
                     case REQ_FAILURE:
+                        LOG_warn << "Error getting file attr";
                         fc->failed(this);
+                        fc->timeout.reset();
                         fc->bt.backoff();
                         fc->urltime = 0;
+                        fc->req.disconnect();
+                        fc->req.status = REQ_PREPARED;
```

Comment:
The same as above. This code ensures that the request is detached from the network layer, that the timeout is deactivated and that the request keep a consistent state when it finishes, mainly to prevent problems with the timeout.